machine_learning_hep/data/database_ml_parameters.yml

```diff
@@ -25,33 +25,27 @@ Dzero:
   - [d_len_xy_ML, imp_par_xy_ML, max_norm_d0d0exp_ML, imp_par_xy_ML, cos_p_ML, d_len_xy_ML, imp_par_xy_ML, cos_p_ML, pt_cand_ML]
   var_others: [inv_mass_ML, pt_cand_ML]
   var_signal: signal_ML
-  bitselvariable : cand_type_ML
+  bitselvariable: cand_type_ML
+  bitselvariable_gen: cand_type_gen_ML
   var_target: signal_ML
   var_training: [d_len_ML, d_len_xy_ML, norm_dl_ML, norm_dl_xy_ML, cos_p_ML, cos_p_xy_ML, imp_par_xy_ML, dca_ML, cos_t_star_ML, imp_par_prod_ML, pt_prong0_ML, pt_prong1_ML, imp_par_prong0_ML, imp_par_prong1_ML, imp_par_err_prong0_ML, imp_par_err_prong1_ML, max_norm_d0d0exp_ML]
   sel_signal: "cand_type_ML>0"
+  sel_signal_gen: "cand_type_gen_ML>0"
```

Comment:
Hi @fcatalan92, this is for the HF analyses indeed a dummy variable, but needed because a 'query' cannot take a null argument (and we want to keep the query(sel_signal) for analyses w/o a bitmap selection). Gian Michele is looking for a better solution, but for the moment this dummy variable selects all candidates and does no harm

Reply:
Ok, now I understand. But are there analysis that use the significance optimization and don't have a bitmap selection? If not, the correspondent "compatibility" code could be dropped at least in `optimization.py`

Reply:
Hi Fabio, Luuk, 
This is is a very good point. I also thought about removing it but then I thought that there are analyses e.g. hypernuclei or HF jets that are using this framework but they are not in the common Ntuplizer, so no bitmap. I also don't like the current solution cause it is prone to errors. E.g. I think the bitmap and the sel_signal/sel_signal_gen should probably be mutually exclusive, to avoid mistakes. Open to suggestions here of course.